src/perf-test-analyis/edfi_perf_test_analysis/volume.py

```diff
@@ -0,0 +1,156 @@
+#     "Page IO Latch Waits started per second",
+# ]
+
+
+def _display_server_counters(results_dir: str, test_type: str, server: str) -> None:
+    markdown(f"### {server.capitalize()} Server Data")
+    file_path = path.join(results_dir, f"{test_type}.ods-3-perf-{server}.csv")
+    if not path.exists(file_path):
+        log_warning("No data file available for display")
+        return
+
+    web = pd.read_csv(file_path)
+    # For regression analysis, need to compare numbers to numbers, not numbers
+    # to dates. Thus "Time" is not appropriate on the x-axis. Copy the index to
+    # a sequence number that can be used instead.
+    web["SequenceNumber"] = web.index
+
+    web = web.astype({"CPU Load (%)": "float64", "Memory Used (%)": "float64"})
+    regression_plot(web, x="SequenceNumber", y="CPU Load (%)")
+    regression_plot(web, x="SequenceNumber", y="Memory Used (%)")
+
+
+def run_analysis() -> Tuple[str, pd.DataFrame]:
+
+    results_dir = select_directory()
+
+    test_type = "volume"
+
+    display_exceptions(results_dir, test_type)
+    display_failures(results_dir, test_type)
+    df_stats = display_stats(results_dir, test_type)
+
+    prepare_for_charts()
+
+    _display_server_counters(results_dir, test_type, "web")
+    _display_server_counters(results_dir, test_type, "db")
+
+    return (results_dir, df_stats)
+
+
+def run_comparison(left_dir: str, df_left: pd.DataFrame) -> None:
+
+    right_dir = select_directory()
+
+    df_right = get_and_prep_stats(right_dir, "volume")
+
+    markdown("## Summary Statistics")
+
+    markdown("### Left Summary")
+    markdown(f"Directory: {left_dir}")
+    display_df(get_summary_stats(df_left))
+
+    markdown("### Right Summary")
+    markdown(f"Directory: {right_dir}")
+    display_df(get_summary_stats(df_right))
+
+    markdown("## Direct Comparison")
+    markdown(
+        "Hypothesis: the average response times for each request are the same in both sets."
+    )
+
+    merged = df_left.merge(df_right, on=["Request"], suffixes=("_l", "_r"))
+
+    mean = "Response Time"
+    stdev = "Approx Std Dev"
+    count = "Request Count"
+
+    merged["diff"] = merged[f"{mean}_l"] - merged[f"{mean}_r"]
+
+    # Perform a Welch's T-Test using the statistics from the CSV files
+    # https://en.wikipedia.org/wiki/Welch%27s_t-test
+    # Should only perform T-Test on samples of 30 or more (Central Limit Theorem)
+    large_enough = merged[(merged[f"{count}_l"] >= 30) & (merged[f"{count}_l"] >= 30)]
+    too_few = merged[(merged[f"{count}_l"] < 30) | (merged[f"{count}_l"] < 30)]
+
+    _, p_value = scipy.stats.ttest_ind_from_stats(
+        mean1=large_enough[f"{mean}_l"],
+        mean2=large_enough[f"{mean}_r"],
+        std1=large_enough[f"{stdev}_l"],
+        std2=large_enough[f"{stdev}_r"],
+        nobs1=large_enough[f"{count}_l"],
+        nobs2=large_enough[f"{count}_r"],
+        equal_var=False,
+    )
+    large_enough.loc[:, "T p-value"] = p_value
```

Comment:
This gives a warning right now in the display, and I haven't yet remembered how to do this correctly. But it does work.

Reply:
Fixed it.